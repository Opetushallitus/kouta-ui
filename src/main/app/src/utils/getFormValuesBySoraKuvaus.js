import mapValues from 'lodash/mapValues';

import parseEditorState from './draft/parseEditorState';

const getFormValuesBySoraKuvaus = soraKuvaus => {
  const {
    nimi,
    julkinen,
    koulutustyyppi,
    kielivalinta,
    metadata = {},
    tila,
  } = soraKuvaus;

  const { kuvaus } = metadata;

  return {
    tila,
    kieliversiot: kielivalinta || [],
    tiedot: {
      nimi: nimi || {},
      kuvaus: mapValues(kuvaus || {}, parseEditorState),
    },
    julkinen: Boolean(julkinen),
    koulutustyyppi: koulutustyyppi || null,
  };
};

export default getFormValuesBySoraKuvaus;
