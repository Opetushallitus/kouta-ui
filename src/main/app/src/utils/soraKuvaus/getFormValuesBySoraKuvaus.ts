import _ from 'lodash/fp';

import { parseEditorState } from '#/src/components/Editor/utils';

const getFormValuesBySoraKuvaus = soraKuvaus => {
  const {
    nimi,
    koulutustyyppi,
    kielivalinta,
    metadata = {},
    tila,
  } = soraKuvaus;

  const { kuvaus } = metadata;

  return {
    tila,
    kieliversiot: kielivalinta || [],
    tiedot: {
      nimi: nimi || {},
      kuvaus: _.mapValues(parseEditorState, kuvaus || {}),
    },
    koulutustyyppi: koulutustyyppi || null,
  };
};

export default getFormValuesBySoraKuvaus;
