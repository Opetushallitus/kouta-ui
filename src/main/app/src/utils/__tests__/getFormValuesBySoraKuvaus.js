import getFormValuesBySoraKuvaus from '../getFormValuesBySoraKuvaus';

test('getFormValuesBySoraKuvaus returns correct form values given sora-kuvaus', () => {
  const values = getFormValuesBySoraKuvaus({
    kieliversiot: ['fi', 'sv'],
    nimi: {
      fi: 'Fi nimi',
      sv: 'Sv nimi',
    },
    metadata: {
      kuvaus: {
        fi: '<b>Kuvaus fi</b>',
        sv: '<i>Kuvaus sv</i>',
      },
    },
    julkinen: true,
    koulutustyyppi: 'amm',
  });

  expect(values).toMatchSnapshot();
});
