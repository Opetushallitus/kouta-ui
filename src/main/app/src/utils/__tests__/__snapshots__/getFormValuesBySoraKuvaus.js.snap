// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFormValuesBySoraKuvaus returns correct form values given sora-kuvaus 1`] = `
Object {
  "julkinen": true,
  "kieliversiot": Array [],
  "koulutustyyppi": "amm",
  "tiedot": Object {
    "kuvaus": Object {
      "fi":         "<p><strong>Kuvaus fi</strong></p>",
      "sv":         "<p><em>Kuvaus sv</em></p>",
    },
    "nimi": Object {
      "fi": "Fi nimi",
      "sv": "Sv nimi",
    },
  },
  "tila": "tallennettu",
}
`;
